### PR TITLE
Unified materialized binary

### PR DESCRIPTION
**Pull request by antiguru**
Compile a unified materialized binary for the emulator

The unified binary links both environmentd and clusterd, and uses the executable name to distinguish its behavior.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

**Review by def- — APPROVED** [review]
Seems to work now, my changes were minimal. Untagged materialized image size: 554.84 MB -> 490.19 MB

**Review by ParkMyCar — APPROVED** [review]
Good idea! LGTM!

File: src/materialized/ci/.gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /clusterd
 /environmentd
```
Comment (benesch) [inline src/materialized/ci/.gitignore:2]:
```suggestion
```

**Review by benesch — APPROVED** [review]
LGTM, thanks! Just one ask: can you add a README or a comment in the crate root for the new materialized crate with a short explanation for why we're doing this?

**Pull request merged**